capreolus/task/rerank.py

```diff
@@ -82,15 +91,18 @@ def rerank_run(self, best_search_run, train_output_path, include_train=False):
             dev_run, self.benchmark.qrels, self.reranker.extractor, relevance_level=self.benchmark.relevance_level
         )
 
+        def local_evaluate_runs(runs):
+            dev_qrels = {qid: self.benchmark.qrels.get(qid, {}) for qid in self.benchmark.folds[fold]["predict"]["dev"]}
+            return evaluator.eval_runs(runs, dev_qrels, evaluator.DEFAULT_METRICS, self.benchmark.relevance_level)
+
         self.reranker.trainer.train(
-            self.reranker,
-            train_dataset,
-            train_output_path,
-            dev_dataset,
-            dev_output_path,
-            self.benchmark.qrels,
-            self.config["optimize"],
-            self.benchmark.relevance_level,
+            reranker=self.reranker,
```

Comment:
Oh will passing keyword arguments here work? The `train()` method doesn't accept kwargs I think - won't this throw an `unexpected keyword argument error`?

Reply:
oh it's always allowed to pass arguments in `name=value` format i think? whether they r keyword arguments or not, but indeed I'm not sure if this follows the pep8 rule

Reply:
It turns out that you can pass a named argument as long as it's the same name used in function definition. Eg:

```
def foo(a, b):
    return a +  b

foo(3, 4)  # works
foo(3, b=4) # works
foo(3, c=4) # throws an error
```

So you are right, this will work here :)